src/macaron/dependency_analyzer/cyclonedx.py

```diff
@@ -35,155 +46,608 @@ def deserialize_bom_json(file_path: Path) -> dict:
 
     Returns
     -------
-    dict
-        The BOM content as a dictionary.
+    Bom
+        The CycloneDX Bom object.
 
     Raises
     ------
     CycloneDXParserError
         If the bom.json file cannot be located or deserialized.
-
     """
-    # TODO: use the official CycloneDX library to validate and deserialize BOM files
-    # once merged: https://github.com/CycloneDX/cyclonedx-python-lib/pull/290
     if not os.path.exists(file_path):
         raise CycloneDXParserError(f"Unable to locate any BOM files at: {str(file_path.parent)}.")
 
+    # We use the `cyclonedx-python-library` library for deserialization following the example here:
+    # https://cyclonedx-python-library.readthedocs.io/en/v7.3.4/examples.html
     with open(file_path, encoding="utf8") as file:
+        json_data = file.read()
+        if defaults.getboolean("dependency.resolver", "validate", fallback=True):
+            try:
+                schema_version = defaults.get("dependency.resolver", "schema", fallback="1.6")
+                my_json_validator = JsonStrictValidator(SchemaVersion.from_version(schema_version))
+            except ValueError as error:
+                raise CycloneDXParserError(f"Unable to find schema validator for {schema_version}: {error}") from error
+            try:
+                validation_errors = my_json_validator.validate_str(json_data)
+            except MissingOptionalDependencyException as error:
+                # Allow MissingOptionalDependencyException.
+                logger.debug("JSON-validation was skipped due to %s", error)
+            except json.JSONDecodeError as error:
+                raise CycloneDXParserError(f"BOM file is invalid: {error}") from error
+
+            if validation_errors:
+                logger.debug("BOM file is invalid: %s", repr(validation_errors))
+                raise CycloneDXParserError(f"BOM file is invalid: {repr(validation_errors)}")
+
+            logger.debug("Successfully validated the BOM file at %s", file_path)
+
         try:
-            return dict(json.load(file))
-        except (ValueError, json.JSONDecodeError) as error:
+            # Mypy complains that `"type[Bom]" has no attribute "from_json"`.
+            # Based on the example provided by cyclonedx-python-lib this type issue needs to be suppressed.
+            # This method is injected into the Bom class that is annotated by ``serializable`` but mypy is not
+            # able to detect that.
+            bom_from_json = Bom.from_json(json.loads(json_data))  # type: ignore[attr-defined]
+        except (ValueError, AttributeError, json.JSONDecodeError) as error:
             raise CycloneDXParserError(f"Could not process the dependencies at {file_path}: {error}") from None
 
+        if isinstance(bom_from_json, Bom):
+            logger.debug("Successfully deserialized the BOM file: %s", repr(bom_from_json))
+            return bom_from_json
```

Comment:
I wonder in which scenario that `bom_from_json` is not of type `Bom`. Is it correct to say `bom_from_json` is not of type `Bom` when no validation happens, and `json_data` is not a valid cyclonedx format :thinking: